### PR TITLE
refactor: rework Tiles removal and cleanup

### PR DESCRIPTION
**Pull request by peppsac**
This commit removes TileMesh specific cleanup code and instead
introduce ObjectRemovalHelper that can be used on any object.

Another change is that attached layers get called on just-removed
elements. This allows each layer to perform its own cleanup.

### CODE REVIEW

**Comment by peppsac** [comment]
I need to fix one last issue and update `TileDebug.js` as well before this is ready.

**Comment by peppsac** [comment]
OK fixed.
You can test the difference by enabling the BoundingBox on the index.html sample and display the geometry count chart.

Then type:
```
globeView.controls.setRange(5000, true)
globeView.controls.setRange(25000000, true)
```
in the console and observe the variation in the geometry count chart.

Ideally, after going back at the initial position the geometry count should be the same as when itowns was started (= 90).

On master, this count never gets back to 90.
This PR fixes this problem.

**Comment by peppsac** [comment]
@gchoqueux r?

File: src/Process/ObjectRemovalHelper.js
Comment (gchoqueux) [inline src/Process/ObjectRemovalHelper.js]: callDispose isn't used

**Review by gchoqueux — CHANGES_REQUESTED** [review]
OK 
just one comment

File: src/Process/ObjectRemovalHelper.js
Comment (peppsac) [reply src/Process/ObjectRemovalHelper.js]: Fixed!

Comment (peppsac) [reply src/Process/ObjectRemovalHelper.js]: For typo like this you can leave a comment but approve the PR.
This way I can fix it and merge directly without waiting for your approval.

**Review by gchoqueux — APPROVED** [review]

**Pull request merged**
